bin/build.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+
+# Builds and uploads:
+#   - CLI to S3
+#   - Spark docker image to dockerhub
+#   - stub universe zip to S3
+#
+# Manifest config:
+#   CLI_VERSION - version label to use for CLI package
+#   SPARK_DIST_URI - where fetch spark distribution from
+#
+# ENV vars:
+#   COMMONS_TOOLS_DIR - path to dcos-commons/tools/, or empty to fetch latest release tgz
+#   DOCKER_IMAGE - "<image>:<version>", falls back to mesosphere/spark-dev:COMMIT)
+#   ghprbActualCommit / GIT_COMMIT - COMMIT value to use for DOCKER_IMAGE, if DOCKER_IMAGE isn't specified
+
+set +e
+
+BIN_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+BASEDIR="${BIN_DIR}/.."
+
+configure_docker_image() {
+    if [ -z "$DOCKER_IMAGE" ]; then
+        # determine image label based on git commit:
+        if [ -n "$ghprbActualCommit" ]; then
+            # always overrides default GIT_COMMIT:
+            export GIT_COMMIT=$ghprbActualCommit
+        fi
+        if [ -z "$GIT_COMMIT" ]; then
+            # Commit not explicitly provided by CI. Fetch directly from Git:
+            export GIT_COMMIT="$(git rev-parse HEAD)"
+        fi
+        if [ -z "$GIT_COMMIT" ]; then
+            echo "Unable to determine git commit. Giving up."
+            exit 1
+        fi
+        export DOCKER_IMAGE="mesosphere/spark-dev:$GIT_COMMIT"
+    fi
+}
+
+fetch_commons_tools() {
+    if [ -z "${COMMONS_TOOLS_DIR}" ]; then
+        pushd ${BIN_DIR}
+        rm -rf dcos-commons-tools/ && curl https://infinity-artifacts.s3.amazonaws.com/dcos-commons-tools.tgz | tar xz
+        popd
+        export COMMONS_TOOLS_DIR=${BIN_DIR}/dcos-commons-tools/
+    fi
+}
+
+notify_github() {
+    ${COMMONS_TOOLS_DIR}/github_update.py $1 build $2
+}
+
+build_cli() {
+    notify_github pending "Building CLI"
```

Comment:
Yes it's for the granular statuses in PRs. In practice its just a convenience for the PR viewer. The upload status is useful in particular because its Details link points to the uploaded stub universe URL which can be passed to `dcos package repo add` by readers to try the build manually.

The GitHub updater script autodetects whether it's in a Jenkins environment (checks for JENKINS_HOME), if it's being called locally it'll automatically just print the status it was passed to stdout, without hitting GitHub.